packages/amplify-graphql-model-transformer/src/graphql-model-transformer.ts

```diff
@@ -323,9 +324,31 @@ export class ModelTransformer extends TransformerModelBase implements Transforme
   };
 
   generateResolvers = (context: TransformerContextProvider): void => {
+    const dataSourceMapping = {};
     this.resourceGeneratorMap.forEach((generator) => {
       generator.generateResources(context);
+      Object.assign(
```

Comment:
Rather than a chain of nested assigns & loops & filters, could you break this out so it's a bit easier to understand the logic here? Reading through this I'm not quite sure what's supposed to be happening in this block.